openmdao/core/vec_wrapper.py

```diff
@@ -943,97 +943,90 @@ def _get_flattened_sizes(self):
-                    if isinstance(alpha_bound, float):
-                        if alpha_bound >= 0.0:
-                            alpha[idx] = min(alpha[idx], alpha_bound)
-                    else:
-                        j = 0
-                        for new_val in alpha_bound:
-                            if new_val >= 0.0:
-                                alpha[idx+j] = min(alpha[idx+j], new_val)
-                            j += 1
-
-                lower = meta.get('lower')
-                if lower is not None:
-                    alpha_bound = (lower - val)/duvec[name]
-                    if isinstance(alpha_bound, float):
-                        if alpha_bound >= 0.0:
-                            alpha[idx] = min(alpha[idx], alpha_bound)
-                    else:
-                        j = 0
-                        for new_val in alpha_bound:
-                            if new_val >= 0.0:
-                                alpha[idx+j] = min(alpha[idx+j], new_val)
-                            j += 1
+                    # If we are already violated for any reason,
+                    # bring it back to the boundary.
+                    if diff < 0.0:
+                        alpha[idx] = alpha_bound
 
-        # Alpha is a scaler
-        else:
-            for name, meta in iteritems(self):
+                    elif alpha_bound >= 0.0:
+                        alpha[idx] = min(alpha[idx], alpha_bound)
 
-                if 'remote' in meta:
-                    continue
+                else:
+                    j = 0
+                    for new_val in alpha_bound:
 
-                val = self[name]
+                        # If we are already violated for any reason,
+                        # bring it back to the boundary.
+                        if diff[j] < 0.0:
+                            alpha[idx+j] = new_val
 
-                upper = meta.get('upper')
-                if upper is not None:
-                    alpha_bound = (upper - val)/duvec[name]
-                    if isinstance(alpha_bound, float):
-                        if alpha_bound >= 0.0:
-                            alpha = min(alpha, alpha_bound)
-                    else:
-                        for new_val in alpha_bound:
-                            if new_val >= 0.0:
-                                alpha = min(alpha, new_val)
-
-                lower = meta.get('lower')
-                if lower is not None:
-                    alpha_bound = (lower - val)/duvec[name]
-                    if isinstance(alpha_bound, float):
-                        if alpha_bound >= 0.0:
-                            alpha = min(alpha, alpha_bound)
-                    else:
-                        for new_val in alpha_bound:
-                            if new_val >= 0.0:
-                                alpha = min(alpha, new_val)
+                        elif new_val >= 0.0:
+                            alpha[idx+j] = min(alpha[idx+j], new_val)
```

Comment:
would be quicker here to just do

if new_val < alpha[idx+j]:
    alpha[idx+j] = new_val

instead of making a function call to min()